airbrakes/state.py

```diff
@@ -146,13 +148,20 @@ def __init__(self, context: "AirbrakesContext"):
     def update(self):
         """Checks to see if the rocket has reached apogee, indicating the start of free fall."""
 
-        # We extend the airbrakes after 1.5 seconds of coasting:
-        if time.time() - self.start_time > AIRBRAKES_AFTER_COASTING and not self.airbrakes_extended:
+        # Check if we are going to overshoot our target apogee, and extend the airbrakes if we are.
+        pred_apogee = self.context.apogee_predictor.apogee
+
+        if pred_apogee is not None and pred_apogee >= TARGET_ALTITUDE:
```

Comment:
Is there any concern of our predicted apogee being wildly high during first few hundred data packets?

Reply:
Yeah its definitely possible, but it's more likely the other way around. The first batch of data points could have some weird noise in it that causes the curve fit to estimate lower drag, which would cause the prediction to shoot high.

Reply:
So it might be wise to still use a time based check on this as well, e.g. check for this condition only after 2 seconds, so it's converged

Reply:
I'm deleting this code for now and we will come back to control in another PR after this is merged. This code won't work because the curve fit is only for data without any airbrakes extension, and as soon as we extend them it will go to shit

Reply:
:(

That will fail the integration test and a bunch of unit tests. Also I thought we don't care about what happens to our predicted apogee after deploying airbrakes

Reply:
okay I'm adding them back purely because I don't want to fix the tests lol

And its more so that it just wont work to control the air brakes doing it that way. As soon as we do that, the accel curve massively changes --> predicted apogee changes --> it's not accurate. We are going to have to write code that predicts the apogee, gets our nice dh vs velocity curve, and then does the control logic.